### ISSUE TITLE
DataPlaneTransfer: Get DataEncrypter from ServiceExtensionContext

### ISSUE THREAD

**Issue by DominikPinsel** [eclipse-dataspaceconnector/DataSpaceConnector#1804]
> As we would like to have a possibility to mitigate this security risk as soon as possible, I'd like to assign this issue to myself to create the corresponding PR.

# Feature Request

At the time of writing the `DataEncrypter` in the `DataPlaneTransferSyncExtension` is hard coded to the `NoopDataEncrypter`. Like this it's not possible to swap out the `NoopDataEncrypter` instance with something that does actual encryption.
I'd propose to make it possible to provide an extension, that implements a secure `DataEncrypter`.

## Which Areas Would Be Affected?

- Data Plane Transfer Sync Extension

## Why Is the Feature Desired?

- Security

## Solution Proposal

**Now**
```java
public void initialize(ServiceExtensionContext context) {
    var keyPair = createKeyPair(context);
    var encrypter = new NoopDataEncrypter();
    var selectorStrategy = context.getSetting(DPF_SELECTOR_STRATEGY, "random");

    // more code
}
```

**Proposal**
```java
@Inject(required = false)
private DataEncrypter encrypter;

@Override
public void initialize(ServiceExtensionContext context) {
    if (encrypter == null) {
        encrypter = new NoopDataEncrypter();
    }

    // more code
}
```

## Type of Issue
improvement

## Checklist

- [x] assigned appropriate label?
- [x] **Do NOT select a milestone or an assignee!**

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

**Comment by ndr-brt** [comment]
The `NoopDataEncryper` could also be provided as default (`@Provider(isDefault = true)`), so that any other extension could swap it with another one

**Comment by DominikPinsel** [comment]
> The `NoopDataEncryper` could also be provided as default (`@Provider(isDefault = true)`), so that any other extension could swap it with another one

Yea, we could do this. Is it ok for an extension, that is no core-extension, to provide a default implementations?
Because the current extension set looks like this
- data-plane-transfer-client
- data-plane-transfer-spi
- data-plane-transfer-sync

**Comment by ndr-brt** [comment]
@DominikPinsel actually there are also other "non-core" extensions that provide default services (as `CosmosClientProviderExtension`, `FederatedCatalogCacheExtension`, ...), so I would say that's ok.

**Comment by paullatzelsperger** [comment]
just wanted to weigh in here: the `isDefault` attribute was originally intended to provide fallback implementations for services, e.g. an `in-memory-store` of any kind, not to realize extensibility. Using it the way it is done here, and other places throughout the code base constitutes the latter. 
So if we want to do things by the book, the `NoopDataEncrypter` should be in its own module/ extension. 

However: 
@ndr-brt will perform major restructuring of the directories and modules (#1803), during which we can move that provider method into some sort of `data-plane-core` extension, similar to what we have in the core now. This will address @DominikPinsel's point:
> Is it ok for an extension, that is no core-extension, to provide default implementations?

because then, there would be the equivalent of `core` in the data plane as well.

TL;DR: we can leave this as-is, because we'll have to comb through the database and double-check all the `isDefault` provider methods later.
